cmd/agent/app/builder_test.go

```diff
@@ -21,27 +21,55 @@
 package app
 
 import (
-	"io/ioutil"
+	"errors"
 	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/uber/jaeger-lib/metrics"
-	"github.com/uber/tchannel-go"
 	"go.uber.org/zap"
 	"gopkg.in/yaml.v2"
 
-	"github.com/uber/jaeger/pkg/discovery"
 	"github.com/uber/jaeger/thrift-gen/jaeger"
 	"github.com/uber/jaeger/thrift-gen/zipkincore"
 )
 
-func TestConfigFile(t *testing.T) {
+var yamlConfig = `
+ignored: abcd
```

Comment:
our config rollout (via puppet) is not synchronized with agent rollout (deb pkg), so it's quite normal to have elements in the config that are read differently by different versions of the agent.